plugins/maciVoting/services/coordinator/coordinator.ts

```diff
@@ -0,0 +1,175 @@
+import { type Address } from "viem";
+import { type z } from "zod";
+import { ESupportedNetworks } from "./utils/networks";
+import { type IGenerateArgs, type ISubmitProofsArgs } from "./utils/types";
+import { getAuthorizationHeader } from "./utils/auth";
+import { getSigner, getPublicClient } from "./utils/chain";
+import { pollDeploymentConfig } from "./config";
+import { GenerateResponseSchema, SubmitResponseSchema } from "./utils/schemas";
+
+const BASE_URL = "http://localhost:3000/v1";
+const CHAIN = ESupportedNetworks.ETHEREUM_SEPOLIA;
+
+type GenerateResponse = z.infer<typeof GenerateResponseSchema>;
+type SubmitResponse = z.infer<typeof SubmitResponseSchema>;
+type CoordinatorServiceResult<T, E = Error> = { success: true; data: T } | { success: false; error: E };
+
+const signer = getSigner(CHAIN);
+const encryptedHeader = await getAuthorizationHeader(signer);
+const publicClient = getPublicClient(CHAIN);
+
+export const merge = async (
+  maciContractAddress: Address,
+  pollId: number,
+  approval: string,
+  sessionKeyAddress: Address
+): Promise<CoordinatorServiceResult<boolean>> => {
+  let response: Response;
+  try {
+    response = await fetch(`${BASE_URL}/proof/merge`, {
```

Comment:
Do not use try/catch like this. You can use it like `fetch(...).catch((error) => ...)`

Reply:
Sure, done. Out of interest, why is there a preference for this way of doing things over try catch?

Reply:
It's just a more functional approach instead of procedural. With `catch` method you don't need to have a mutation for `response` and keep in `const`, and it makes code cleaner because you can't `return` the value from `catch` method so you need to explicitly handle it outside which makes it more obvious when you read the code (not a good practice to have some logic handling in catch).

Reply:
> you don't need to have a mutation for response and keep in const

I like this benefit

> makes code cleaner because you can't return the value from catch method so you need to explicitly handle it outside which makes it more obvious when you read the code (not a good practice to have some logic handling in catch).

Ok yeah I see how this can be cleaner

Thanks for the explanation